src/thread.cpp

```diff
@@ -158,10 +178,39 @@ void zmq::thread_t::setSchedulingParameters(int priority_, int schedulingPolicy_
 #endif
 
     posix_assert (rc);
+
+    if (use_nice_instead_priority &&
+            thread_priority != ZMQ_THREAD_PRIORITY_DFLT)
+    {
+        // assume the user wants to decrease the thread's nice value
+        // i.e., increase the chance of this thread being scheduled: try setting that to
+        // maximum priority.
+        rc = nice(-20);
+
+        // IMPORTANT: ignore return code: EPERM is typically returned for unprivileged processes
+        //            (CAP_SYS_NICE capability is required or RLIMIT_NICE resource limit should be set)
+        //            but in that case there is no much we can do
+
+    }
+
+    if (thread_affinity != ZMQ_THREAD_AFFINITY_DFLT)
+    {
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        for (unsigned int cpuidx=0; cpuidx<sizeof(int)*8; cpuidx++)
+        {
+            int cpubit = (1 << cpuidx);
+            if ( (thread_affinity & cpubit) != 0 )
+                CPU_SET( cpuidx , &cpuset );
+        }
+        rc = sched_setaffinity(0 /* use calling thread PID */, sizeof(cpu_set_t), &cpuset);
```

Comment:
```
(If you are using the POSIX  threads
       API, then use pthread_setaffinity_np(3) instead of sched_setaffinity().)
```

Also given it's non-posix, you'll need to add checks in aclocal.m4 and CMakeList.txt and IFDEF it out if not available, otherwise the build on non-linux system will be broken